### PR TITLE
Addition of engine rpc-types from reth

### PR DESCRIPTION
**Pull request by 1010adigupta**
## Motivation
This pr aims to add engine related rpc-types to alloy.
closes paradigmxyz/reth#5989
## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types/src/eth/transaction/request.rs [hunk 1]
```diff
@@ -62,6 +64,17 @@ impl From<OptimismTransactionFields> for OtherFields {
...
+#[derive(Clone, Debug, PartialEq, Eq, Default, Serialize, Deserialize)]
+#[repr(C)]
+pub struct BlobTransactionSidecar {
+    /// The blob data.
+    pub blobs: Vec<Blob>,
+    /// The blob commitments.
+    pub commitments: Vec<Bytes48>,
+    /// The blob proofs.
+    pub proofs: Vec<Bytes48>,
+}
```
Comment (1010adigupta) [inline crates/rpc-types/src/eth/transaction/request.rs:77]: I have added BlobTransactionSidecar here which is the type for sidecar field for EIP-4844 transactions, in TransactionRequest, is this the correct approach or should I do this using other::OtherFields?

**Review by mattsse — CHANGES_REQUESTED** [review]
great draft!

I'd all of this in a new crate rpc-engine-types

**Review by mattsse — APPROVED** [review]
nice job!

now we can replace this in reth-rpc-types and simply reexport everything in the engine module there

File: crates/rpc-engine-types/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# ethereum
+alloy-rlp = { workspace = true, features = ["arrayvec", "derive"] }
+alloy-primitives = { workspace = true, features = ["rlp", "serde"] }
+alloy-rpc-types.workspace = true
+
+serde = { workspace = true, features = ["derive"] }
+thiserror.workspace = true
+
+# arbitrary
+arbitrary = { version = "1.3", features = ["derive"], optional = true }
```
Comment (Evalir) [inline crates/rpc-engine-types/Cargo.toml:24]: Do we need arbitrary?

**Review by Evalir — APPROVED** [review]
lgtm! thanks! just wanna make sure we're not copy extra unused deps

**Comment by 1010adigupta** [comment]
> nice job!
> 
> now we can replace this in reth-rpc-types and simply reexport everything in the engine module there

on it!
should I create a issue in reth for this, and attach a new pr there now?

**Pull request merged**